streams/src/test/java/org/apache/kafka/streams/kstream/KStreamBuilderTest.java

```diff
@@ -77,23 +77,41 @@ public void testNewName() {
         assertEquals("Y-0000000001", builder.newName("Y-"));
         assertEquals("Z-0000000002", builder.newName("Z-"));
 
-        KStreamBuilder newBuilder = new KStreamBuilder();
+        final KStreamBuilder newBuilder = new KStreamBuilder();
 
         assertEquals("X-0000000000", newBuilder.newName("X-"));
         assertEquals("Y-0000000001", newBuilder.newName("Y-"));
         assertEquals("Z-0000000002", newBuilder.newName("Z-"));
     }
 
+
+    @Test
+    public void shouldHaveSinkTopicForStreamTo() {
+        final KStream<String, String> source = builder.stream("topic-source");
+        source.to("topic-sink");
+
+        final MockProcessorSupplier<String, String> processorSupplier = new MockProcessorSupplier<>();
+
+        source.process(processorSupplier);
+
+        driver = new KStreamTestDriver(builder);
+        driver.setTime(0L);
+
+        driver.process("topic-source", "A", "aa");
```

Comment:
@wimvanleuven @mjsax A few questions for me to understand the PR:

My understanding is that in your test case, you want to call sth. like `process("topic-sink")` which will give you an NPE, hence the below changes in `KStreamTestDriver.java`. However, the added unit test added a sink node but did not process from it, so it seems not covering the problem.

In addition, I'm not sure what's the motivation to get the sink node in `process()` and then pass the record to its children, which should be empty. Could you guys elaborate a bit more?